### DISCUSSION TITLE
Custom screen integration

### DISCUSSION THREAD

**Discussion by timetofly06** [forntoh/LcdMenu#85] [Ideas]
Hello 

Until now, I was using lcd.setCursor(x,y) and lcd.print methods from traditionnal LiquidCrystal_I2C library to build custom screens with measurements values. I didn't see any option to integrate customs screens but may be I missed it

It would be a new item class, let's call it "ItemScreen". it will act like a submenu, but it would display a custom screen.
The custom screen (composed with lcd.print and lcd.setcursor would be described as a function in the user program and declared as extern in order to be called by library LcdMenu.h

Hope you'll be interested with my proposal ;-)

**Comment by timetofly06** [comment]
I'm answering to myself : 

I added another inclusion of LiquidCrystal_I2C.h directly to my main program and add a new object "screen" to adress it : 

#include <LiquidCrystal_I2C.h>
LiquidCrystal_I2C screen(0x27,20,4);  // set the LCD address to 0x27 for a 16 chars and 2 line display

and init it : 

    screen.init();  //initialize the lcd
    screen.backlight();  //open the backlight 

Then, I added a new command object from the main menu : 

                    ItemCommand("INFOS", mesures),

and finally complete with the called function "mesures" where I can set the LCD commands : 

void mesures() {
  screen.clear();
  screen.setCursor ( 0, 0 ); 
  screen.print("hh:mm:ss");
  delay(5000);                           // just for test - Now I have to add a while loop and break it when "back key" is pressed
}

OK. So it works, but  it's not optimized, as I have two display objects (screen and lcd). So obviously it's not a "Beautiful code" at all, haha !

It would be cool to get this feature in a future version ;-)

**Comment by forntoh** [comment]
You can use `lcd` from `LcdMenu`, there's a `hide` function.

e.g.
```cpp
...
menu.hide();
menu.lcd->setCursor(0, 0);
menu.lcd->print("hh:mm:ss");
...
```

Hide function signature

```cpp
/**
 * When you want to display any other content on the screen then
 * call this function then display your content, later call
 * `show()` to show the menu
 */
void hide();
```

**Reply by timetofly06** [reply]
So great ! I didn't see the trick.  You're a boss ! ;-)
